src/Datasets.php

```diff
@@ -42,12 +42,69 @@ public static function set(string $name, $data): void
     public static function get(string $name)
     {
         if (!array_key_exists($name, self::$datasets)) {
-            throw new DatasetDoesNotExist($name);
+            return self::extractDatasetKeys($name);
         }
 
         return self::$datasets[$name];
     }
 
+    public static function unset(string $name): void
+    {
+        if (array_key_exists($name, self::$datasets)) {
+            unset(self::$datasets[$name]);
+        }
+    }
+
+    /**
+     * @return array[]
+     */
+    private static function extractDatasetKeys(string $datasetName): array
```

Comment:
Else try splitting the method into smaller methods. 

Or potentially having a whole class for resolving datasets with keys 😄

Reply:
I've refactored the code, used an `ExtractsDatasetKeys` trait instead of a brand new helper class, I think this is more semantically consistent

additionally, Both Datasets than the helper class would have cross references between them, did not like it so much

what do you think?